### ISSUE TITLE
Error compiling fastQValidator

### ISSUE THREAD

**Issue by denisemauldin** [statgen/libStatGen#12]
Compiling libStatGen doesn't throw any errors, but when compiling fastQValidator, it does:

> g++  -std=c++0x -O4 -pipe -Wall  -I/tools/src/libStatGen//include -I.   -D__ZLIB_AVAILABLE__ -D_FILE_OFFSET_BITS=64 -D__STDC_LIMIT_MACROS   -o ../bin/fastQValidator     ../obj/FastQValidator.o      /tools/src/libStatGen//libStatGen.a     -lm -lz  
> /tools/src/libStatGen//libStatGen.a(GzipFileType.o): In function `GzipFileType::GzipFileType(char const*, char const*)':
> GzipFileType.cpp:(.text+0xdf): undefined reference to `gzopen64'
> /tools/src/libStatGen//libStatGen.a(GzipFileType.o): In function `GzipFileType::seek(long, int)':
> GzipFileType.cpp:(.text._ZN12GzipFileType4seekEli[GzipFileType::seek(long, int)]+0x9): undefined reference to `gzseek64'
> /tools/src/libStatGen//libStatGen.a(GzipFileType.o): In function `GzipFileType::tell()':
> GzipFileType.cpp:(.text._ZN12GzipFileType4tellEv[GzipFileType::tell()]+0x5): undefined reference to `gztell64'

**Comment by mktrost** [comment]
Do you have the development version of zlib installed?

**Comment by denisemauldin** [comment]
Yes.  1.2.8.  I was incorrect though and it's libStatGen that's throwing the error.  The compile doesn't show anything, but the tests fail.   I had to manually add the -L/tools/lib directory containing -lz to the Makefiles/Makefile.include. Not sure if there's something you can do to detect that.  It is in my LD_LIBRARY_PATH.

**Comment by jonathonl** [comment]
On which OS are you building?

**Comment by denisemauldin** [comment]
CENTOS.  Mostly I think it's that I have my updated libraries in a different path (/tools) rather than in the system paths.

**Comment by jonathonl** [comment]
So adding "-L/tools/lib" solved your problem then? Or are you still getting errors when building the tests?

**Comment by denisemauldin** [comment]
Yes, adding -L/tools/lib solved my problem. :)

**Comment by sam0per** [comment]
Hi @denisemauldin 
I am encountering a very similar problem and following your message thread, I tried to modify the libStatGen/Makefiles/Makefile.include. Particularly, I changed this line:

USE_ZLIB = -D__ZLIB_AVAILABLE__
into
USE_ZLIB = -L/usr/local/packages

and this line:
ZLIB_LIB = -lz
into
ZLIB_LIB = -lz -L/usr/local/packages

I also tried to combine different option but I could not make it work.
Where exactly did you add "-L/tools/lib" which I suppose contains the reference to the _zlib_ package
